src/traces/scatter/hover.js

```diff
@@ -107,6 +107,11 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
         }
 
         if(inside) {
+            // constrain ymin/max to the visible plot, so the label goes
+            // at the middle of the piece you can see
+            ymin = Math.max(ymin, 0);
+            ymax = Math.min(ymax, ya._length);
```

Comment:
Great. That sounds good. 

Now that all the scatter ternary hover label testing infrastructure is in place, would you mind adding a test case for this?